### PR TITLE
Conditional-Signing Example Rework

### PR DESCRIPTION
**Pull request by awisniew207**
Reworked the `conditional-signing` examples:
- JavaScript to TypeScript
- Old networks to DatilTest
- Implemented Payments
- Updated READMEs
- Changed `browser` example from Parcel to Vite
- Updated the Lit Actions and Styilized the implementations

### CODE REVIEW

File: conditional-signing/browser/README.md [hunk 1]
```diff
@@ -8,11 +8,11 @@ This code example demonstrates how to execute a Lit Action that only provides a
 
 In this directory, `conditional-signing/browser`, run `yarn` to install the project dependencies.
 
-### Specifying Your Lit PKP's Public Key
+### Specifying Your Lit PKP's Public Key and CapacityCredit Token ID
 
-If you already have a Lit PKP that you'd like to use, you can copy the contents of the provided `.env.example` to a `.env` file to specify it. If you don't have a PKP, or wish to use a new one for this example, then you can skip this step and one will be created for you when you run this example.
+If you already have a Lit PKP that you'd like to use, you can copy the contents of the provided `.env.example` to a `.env` file to specify it. If you don't have a PKP, or wish to use a new one for this example, then you can skip this step and one will be created for you when you run this example. This is also the case if you'd like to use a different Lit Capacity Credit Token ID.
```
Comment (spacesailor24) [inline conditional-signing/browser/README.md:13]:
```suggestion
If you already have a Lit PKP that you'd like to use, you can copy the contents of the provided `.env.example` to a `.env` file to specify it. If you don't have a PKP, or wish to use a new one for this example, then you can skip this step and one will be created for you when you run this example. This is also the case for the Lit Capacity Credit Token ID; You can provide one or have one minted for you.
```

File: conditional-signing/browser/README.md [hunk 2]
```diff
@@ -21,50 +21,46 @@ cp .env.example .env
...
+VITE_LIT_CAPACITY_CREDIT_TOKEN_ID=yourCapacityCreditTokenId
+VITE_CHAIN_TO_CHECK_CONDITION_ON=ethereum
 ```
 
+**NOTE** By changing the `VITE_CHAIN_TO_CHECK_CONDITION_ON` value, you can change the chain that the Lit Action will check the condition on. The current `.example.env` file sets this value to `ethereum`.
+
 ### Starting the Example
 
-In this directory, `conditional-signing/browser`, run `yarn start` to bundle all of this code and server the HTML file at: [http://localhost:1234](http://localhost:1234).
+In this directory, `conditional-signing/browser`, run `yarn dev` to bundle all of this code and server the HTML file at: [http://localhost:5173](http://localhost:5173).
```
Comment (spacesailor24) [inline conditional-signing/browser/README.md:33]:
```suggestion
In this directory, `conditional-signing/browser`, run `yarn dev` to bundle all of this code and serve the HTML file at: [http://localhost:5173](http://localhost:5173).
```

File: conditional-signing/browser/src/connect.ts
Comment (spacesailor24) [inline conditional-signing/browser/src/connect.ts:1]: Can you rename this file to `conditionalSigning`

File: conditional-signing/browser/src/connect.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+import { litActionCode } from "./litAction";
+import { getEnv } from "./utils";
+
+const LIT_PKP_PUBLIC_KEY = getEnv("VITE_LIT_PKP_PUBLIC_KEY");
+const LIT_CAPACITY_CREDIT_TOKEN_ID = getEnv("VITE_LIT_CAPACITY_CREDIT_TOKEN_ID");
+const CHAIN_TO_CHECK_CONDITION_ON = getEnv("VITE_CHAIN_TO_CHECK_CONDITION_ON");
+const LIT_NETWORK = LitNetwork.DatilTest;
+
+export const conditionalSigning = async() => {
+  console.log("Clicked");
```
Comment (spacesailor24) [inline conditional-signing/browser/src/connect.ts:23]:
```suggestion
```

File: conditional-signing/browser/src/connect.ts [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    ethAddress?: string;
+  } = {
+    publicKey: LIT_PKP_PUBLIC_KEY,
+  };
+
+  try {
+    const provider = new ethers.providers.Web3Provider(window.ethereum);
+    await provider.send("eth_requestAccounts", []);
+    const ethersSigner = provider.getSigner();
+    console.log("ethersSigner:", ethersSigner);
```
Comment (spacesailor24) [inline conditional-signing/browser/src/connect.ts:36]:
```suggestion
```
Probably don't need this console log?

File: conditional-signing/browser/src/connect.ts [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+    console.log("✅ Connected to the Lit network");
+
+    console.log("🔄 Connecting LitContracts client to network...");
+    const litContracts = new LitContracts({
+      signer: ethersSigner,
+      network: LIT_NETWORK,
+      debug: false,
+    });
+    await litContracts.connect();
+    console.log("✅ Connected LitContracts client to network");
```
Comment (spacesailor24) [inline conditional-signing/browser/src/connect.ts:54]:
```suggestion
    console.log("✅ Connected LitContracts client to network");

```

File: conditional-signing/browser/src/connect.ts [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+            resources: [
+              {
+                resource: new LitPKPResource("*"),
+                ability: LitAbility.PKPSigning,
+              },
+              {
+                resource: new LitActionResource("*"),
+                ability: LitAbility.LitActionExecution,
+              },
+            ],
```
Comment (spacesailor24) [inline conditional-signing/browser/src/connect.ts:164]:
```suggestion
            resources: [],
```
I don't think you need to provide any permissions for this sig since we're just using it to authenticate an address

File: conditional-signing/browser/src/connect.ts [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+            method: "eth_getBalance",
+            parameters: [":userAddress", "latest"],
+            returnValueTest: {
+              comparator: ">=",
+              value: "1",
+            },
+          },
+        ],
+        authSig: await (async () => {
+          const toSign = await createSiweMessageWithRecaps({
```
Comment (spacesailor24) [inline conditional-signing/browser/src/connect.ts:150]: Can probably just use [createSiwe](https://v6-api-doc-lit-js-sdk.vercel.app/functions/auth_helpers_src.createSiweMessage.html)

File: conditional-signing/browser/src/utils.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+export const getEnv = (name: string) => {
```
Comment (spacesailor24) [inline conditional-signing/browser/src/utils.ts:1]: The point of this function is throw an error if required ENVs are passed. If the ENV isn't required, then you can just use `import.meta.env[name]`. As is this function is not different than just using `import.meta.env[name]`

File: conditional-signing/nodejs/README.md [hunk 9]
```diff
@@ -18,66 +18,78 @@ cp .env.example .env
...
+   ```
+    ℹ️  PKP token ID: 0x0ef9cdbcae80b86092c4dd874ee2718ffbbbddef662d437efc36c66ad5db45d3
+    ℹ️  PKP public key: 04240f3dac4fb3431c0df0839581ee45dbb1be1d92d4d74355857f9953c104f36df6725e70897f9fe97f66f783d9e0a7287ba4f8fd023bffca701037331e17ccf6
+    ℹ️  PKP ETH address: 0xa6DF77506934c49211787C643f345Ea7B68a9773
+   ```
+5. Next, depending on whether or not you specified a CapacityCredit Token ID in the `.env` file:
+   - If you did, you'll be on the next step
+   - If you didn't, your provided private key will be used to sign a transaction to mint a new CapacityCredit on the Lit Chronicle Yellowstone rollup. Agin, this transaction will cost `tstLPX`. Your CapacityCredit token ID will be `console.log`ed, you can search the console output for: `✅ CapacityCredit successfully minted` and you should see something similar to:
```
Comment (spacesailor24) [inline conditional-signing/nodejs/README.md:60]:
```suggestion
   - If you didn't, your provided private key will be used to sign a transaction to mint a new CapacityCredit on the Lit Chronicle Yellowstone rollup. Again, this transaction will cost `tstLPX`. Your CapacityCredit token ID will be `console.log`ed, you can search the console output for: `✅ CapacityCredit successfully minted` and you should see something similar to:
```

File: conditional-signing/nodejs/package.json [hunk 10]
```diff
@@ -6,17 +6,27 @@
...
   "dependencies": {
     "@dotenvx/dotenvx": "^0.37.1",
-    "@lit-protocol/auth-helpers": "^6.2.2",
-    "@lit-protocol/constants": "^6.2.2",
-    "@lit-protocol/contracts-sdk": "^6.2.2",
-    "@lit-protocol/lit-node-client-nodejs": "^6.2.2",
+    "@lit-protocol/auth-helpers": "^6.4.4",
+    "@lit-protocol/constants": "^6.4.4",
+    "@lit-protocol/contracts-sdk": "^6.4.4",
+    "@lit-protocol/lit-node-client-nodejs": "^6.4.4",
```
Comment (spacesailor24) [inline conditional-signing/nodejs/package.json:16]: Can we update the version here and for the browser example to `6.4.10`

File: conditional-signing/nodejs/src/index.ts [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+export const conditionalSigning = async () => {
+  let litNodeClient: LitNodeClientNodeJs;
+  let pkpInfo: {
+    tokenId?: string;
+    publicKey?: string;
+    ethAddress?: string;
+  } = {
+    publicKey: LIT_PKP_PUBLIC_KEY,
+  };
+
```
Comment (spacesailor24) [inline conditional-signing/nodejs/src/index.ts:31]:
```suggestion
```

File: conditional-signing/nodejs/src/index.ts [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+            method: "eth_getBalance",
+            parameters: [":userAddress", "latest"],
+            returnValueTest: {
+              comparator: ">=",
+              value: "1",
+            },
+          },
+        ],
+        authSig: await (async () => {
+          const toSign = await createSiweMessageWithRecaps({
```
Comment (spacesailor24) [inline conditional-signing/nodejs/src/index.ts:149]: Switch to `createSiwe`

File: conditional-signing/nodejs/src/index.ts [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+            nonce: await litNodeClient.getLatestBlockhash(),
+            resources: [
+              {
+                resource: new LitPKPResource("*"),
+                ability: LitAbility.PKPSigning,
+              },
+              {
+                resource: new LitActionResource("*"),
+                ability: LitAbility.LitActionExecution,
+              },
```
Comment (spacesailor24) [inline conditional-signing/nodejs/src/index.ts:162]: Remove these

File: conditional-signing/nodejs/src/index.ts [hunk 14]
```diff
@@ -0,0 +1,185 @@
...
+            toSign,
+          });
+        })(),
+        chain: CHAIN_TO_CHECK_CONDITION_ON,
+        dataToSign: ethers.utils.arrayify(ethers.utils.keccak256([1, 2, 3, 4, 5])),
+        publicKey: pkpInfo.publicKey,
+      },
+    });
+    console.log("✅ Lit Action executed successfully");
+    console.log(litActionSignatures);
```
Comment (spacesailor24) [inline conditional-signing/nodejs/src/index.ts:177]:
```suggestion
```

I don't think we need to log these since they're large and displace the rest of the logs significantly

File: conditional-signing/nodejs/src/utils.ts
Comment (spacesailor24) [inline conditional-signing/nodejs/src/utils.ts:1]: Revert the changes to this function. This function should only be used for required ENVs, otherwise just use `process.env[name]`

File: conditional-signing/nodejs/test/index.spec.ts
Comment (spacesailor24) [inline conditional-signing/nodejs/test/index.spec.ts:1]: I think the two response schemas should be two different tests. For the `address does not have 1 or more Wei` test you can just generate a new private key using ethers to ensure if doesn't have a balance. For the other test it's okay if it fails if the signing account doesn't hold ether. When we add this to be ran by the CI we'll just switch the network to Yellowstone and use a funded key

File: conditional-signing/nodejs/src/index.ts (hunk 13 above)
Comment (awisniew207) [reply conditional-signing/nodejs/src/index.ts:162]: `CreateSiweMessage` expects some resource, leaving an empty array makes it mad. Also I believe we need the Lit Action execution resource?

File: conditional-signing/browser/src/connect.ts (hunk 6 above)
Comment (awisniew207) [reply conditional-signing/browser/src/connect.ts:164]: I think Lit Action execution is required

File: conditional-signing/nodejs/test/index.spec.ts
Comment (awisniew207) [reply conditional-signing/nodejs/test/index.spec.ts:1]: Changed the `conditionalSigning` function definition a little:
```
export const conditionalSigning = async (privateKey?: string) => {
  const ETHEREUM_PRIVATE_KEY = privateKey || getEnv("ETHEREUM_PRIVATE_KEY");
 ```
 
 Changed the test case a little:
 ```
   it("Should fail with an unfunded account", async () => {
    const wallet = ethers.Wallet.createRandom();
    const signedTx = await conditionalSigning(wallet.privateKey);
    expect(signedTx).to.be.jsonSchema(conditionalSigningResponseSchemaNotFunded);
  }).timeout(100_000);
 ```
 
 Let me know if this is what you had in mind!

**Review by spacesailor24 — APPROVED** [review]
